src/MainWindow.cpp

```diff
@@ -83,9 +83,51 @@ QAction *MainWindow::createAction(const QIcon &icon, const QString &text, const
 
 void MainWindow::newFile()
 {
-    // TO-DO: Implement new file function
-}
+    // First time save and editor is not empty
+    if (!this->editor->toPlainText().isEmpty() && currentFileName.isEmpty())
+    {
+        MainWindow::saveFile();
+        // Check that the file has been saved properly
+        if (currentFileName.isEmpty())
+        {
+            return;
+        }
+    }
+    // Check if file has been previously saved
+    else if (!currentFileName.isEmpty())
+    {
+        // Read from saved file and compare to current file
+        QFile file(currentFileName);
 
+        if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+            return;
+        QTextStream in(&file);
+        QString savedFileContents = in.readAll();
+        if (savedFileContents != this->editor->toPlainText().trimmed())
+        {
+            // Create box to prompt user to save changes to file
+            QMessageBox promptBox;
+            promptBox.setWindowTitle("Changes Detected");
+            promptBox.setText("Would you like to save the current changes to the file?");
+            promptBox.setStandardButtons(QMessageBox::Save | QMessageBox::Cancel);
+            promptBox.setDefaultButton(QMessageBox::Save);
+            int option = promptBox.exec();
+
+            // return if the user hit Cancel button
+            if (option == QMessageBox::Cancel)
+            {
+                return;
+            }
+
+            MainWindow::saveFile();
+        }
+    }
+
+    // New window will be created with the untitled name
+    MainWindow *newWindow = new MainWindow();
+    newWindow->setWindowTitle("Code Astra ~ untitled");
+    newWindow->show();
+}

```

Comment:
add a space between the function close curly bracket, and the header of the next function.